crunchers/cruncher.h

```diff
@@ -136,6 +138,63 @@ template <typename IMPL, size_t BUFFER_SIZE = 1024 * 1024>
 using StreamCruncher =
     current::ss::StreamSubscriber<GenericCruncherImpl<IMPL, BUFFER_SIZE>, typename IMPL::event_t>;
 
+// An extension of the `GenerucCruncherImpl` wrapper above,
+// which also generates "ticks" at specified time intervals.
+// The IMPL has the same requirements except it must also implement `OnTick()` method.
+template <typename IMPL, size_t BUFFER_SIZE>
+struct GenericTickCruncherImpl : public GenericCruncherImpl<IMPL, BUFFER_SIZE> {
```

Comment:
* s/GenericTickCruncherImpl/GenericCruncherWithTicksImpl/?
* Public inheritance is default for structs. Maybe you want to make them classes?